### DISCUSSION TITLE
blog/alphanet/

### DISCUSSION THREAD

**Discussion by giscus** [jeremy-feng/giscus#17] [Announcements]
# blog/alphanet/

借鉴卷积神经网络的思想，通过自定义运算符函数，构造类似卷积层的特征提取层。结合批标准化层、池化层、全连接层，搭建 AlphaNet-V1，实现从量价数据到收益率预测的自动挖掘。

https://fengchao.pro/blog/alphanet/

**Comment by JiejiangWu** [comment]
不错的尝试，但个人经验：nn的最重要能力来自于特征提取，在cnn中即是卷积层。文字把卷积层的参数fix，只优化bn层和线性层参数，nn的能力相当于缺失了大部分，结果也符合预期。

**Reply by jeremy-feng** [reply]
谢谢你的讨论~

是的，固定卷积层的参数，相当于指定了特征提取的运算符（当然，这些运算符确实是手工构造因子时比较常用的，因此使用它们是有道理的）。神经网络的工作只不过是在将这些运算符的结果与标签进行拟合。

**Comment by Ydd150** [comment]
本人小白，看了这一篇尝试深有感触，想请问下大佬，训练集的loss明显高于你验证集的loss,这是否有不妥之处啊？

**Reply by jeremy-feng** [reply]
你好，我当时做这个项目的时候用的特征是很基础的字段，标签是未来 10 日收益率，也没有进行什么预处理，并且由于计算资源有限，只选取了少量样本，最终的训练效果其实不算好。

关于 Loss 值的问题，我认为训练集的 Loss 不是必然低于验证集的 Loss 的。有可能刚好训练集的标签整体上波动较大或极值较多，因此较难预测，导致计算出的 Loss 自然偏大。或许还有其他原因，总之我觉得不必纠结 Loss 值的大小，更重要的可能是关注 Loss 值的变化趋势。

**Reply by Ydd150** [reply]
大佬，我一计算机的小白，想请教下有何入门推荐，，比如我随便构造了一个因子，然后我应该干什么，怎么去测试这个因子的效果好不好，有哪些测试方法，大佬能给说说吗？

**Reply by jeremy-feng** [reply]
你可以阅读一下各家券商金融工程组的研报，其中会展示许多常用的因子评价方法，例如 IC、ICIR、分组收益、多空收益、多空信息比率、换手率等，以及各种辅助分析因子绩效的图表。

也推荐一本更加学术的书籍：石川老师的《因子投资——方法与实践》，尤其是其中的第二章和第七章，比较系统地介绍了因子投资的方法论。

**Reply by Ydd150** [reply]
谢谢大佬，方便加您个微信吗？过程中遇到问题想在向您请教，wxidydd12311410，大佬，这是我的微信号

**Reply by jeremy-feng** [reply]
不客气，欢迎交流～可以通过邮件联系，在关于页面有我的邮箱。
